driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -75,17 +135,24 @@ fn serialize_balances(state: &models::AccountState, orders: &[models::Order]) ->
                 token_balance
             });
     }
-    json!(accounts)
+    accounts
+}
+
+fn serialize_order(order: &models::Order, id: &str) -> solver_input::Order {
+    solver_input::Order {
+        account_id: account_id(order.account_id),
+        sell_token: token_id(order.sell_token),
+        buy_token: token_id(order.buy_token),
+        sell_amount: order.sell_amount.to_string(),
+        buy_amount: order.buy_amount.to_string(),
+        id: id.to_owned(), //TODO this should not be needed
```

Comment:
Also a good question for @twalth3r: Is "ID" used anywhere in the solver for any reason?

Reply:
The solver accesses orders through an ID. Mainly because it does stuff with them, e.g., aggregate/modify them, and needs a way to keep track of them throughout the process.

Reply:
I would also advocate for using the id field for the actual order Id stored on the EVM. This is an index of the individual users orders and wouldn't necessarily be unique.

Reply:
> This is an index of the individual users orders and wouldn't necessarily be unique

This is why you can't use it. But you could make it unique by concatenating with address.

Reply:
The solver uses the ID if it is there, otherwise generates a hash from the order data.